### DISCUSSION TITLE
E828: Cannot open undo file for writing

### DISCUSSION THREAD

**Discussion by RigottiG** [LazyVim/LazyVim#1324] [Q&A]
![Screenshot from 2023-08-11 16-02-56](https://github.com/LazyVim/LazyVim/assets/13246770/1a64f45a-417e-4eb9-a1cb-a7a247cf958f)

whenever I save a file, I get this error on the screen, I am with the initial configuration of lazyvim, without changes.

**Comment by dpetka2001** [comment]
What is the output of `ls -al ~/.local/state/nvim/` and `ls -al ~/.local/state/nvim/undo/file_path_from_error` ?

**Comment by ipeglin** [comment]
Was there eventually a fix for this issue? I am struggling with the same problem, and cannot seem to find other related (and resolved) posts.

**Reply by archcorsair** [reply]
I ran into this same issue, the cause was `~/.local/state/nvim/undo/` changed ownership to `root` instead of my user.
Assuming you're also on macOS you can fix this this by running: `sudo chown -R <your user>:staff ~/.local/state/nvim/undo`

**Reply by Andrew15-5** [reply]
I debugged this issue but in `mhartington/formatter.nvim` and it appears that it's simply an FS limitation of 255 bytes. So for very long and deep file names it gives me this error. And hence why it doesn't save any undo data on its own. Gotta find out if I can just make a hierarchy instead of using `%` for `/` in a long file name.
